addons/base_automation/models/base_automation.py

```diff
@@ -853,27 +853,28 @@ def patch(model, name, method):
                     self.env.registry.clear_cache('templates')
 
             if automation_rule.model_id.is_mail_thread and automation_rule.trigger in MAIL_TRIGGERS:
-                def _message_post(self, *args, **kwargs):
-                    message = _message_post.origin(self, *args, **kwargs)
-                    # Don't execute automations for a message emitted during
-                    # the run of automations for a real message
-                    # Don't execute if we know already that a message is only internal
-                    message_sudo = message.sudo().with_context(active_test=False)
-                    if "__action_done"  in self.env.context or message_sudo.is_internal or message_sudo.subtype_id.internal:
-                        return message
-                    if message_sudo.message_type in ('notification', 'auto_comment', 'user_notification'):
-                        return message
-
-                    # always execute actions when the author is a customer
-                    mail_trigger = "on_message_received" if message_sudo.author_id.partner_share else "on_message_sent"
-                    automations = self.env['base.automation']._get_actions(self, [mail_trigger])
-                    for automation in automations.with_context(old_values=None):
-                        records = automation._filter_pre(self)
-                        automation._process(records)
+                def make_message_post():
```

Comment:
it doesn't really matter, but it would be cleaner to have `make_message_post` outside of the loop

otherwise we also redefine make_message_post at each iteration, while we just needed to call it several times not redefine it several times (both version are ok but the current one is a little superfluous)